### PR TITLE
Add Nexus `failure_reason` metric tag

### PR DESCRIPTION
**Pull request by bergundy**
A follow up for #1664 where we decided to add more information on Nexus task failures.

### CODE REVIEW

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Comment by Quinn-With-Two-Ns** [comment]
On the test failure `TestNonDeterminismFailureCauseReplay` is flaky , @yuandrew is looking into it

File: internal/common/metrics/tags.go [hunk 1]
```diff
@@ -82,6 +82,13 @@ func NexusTags(service, operation, taskQueueName string) map[string]string {
 	}
 }
 
+// NexusTaskFailureTags returns a set of tags for Nexus Operation failures.
+func NexusTaskFailureTags(reason string) map[string]string {
+	return map[string]string{
+		FailureReasonTagName: reason,
```
[the remark this answers is not in the document]
Comment (cretz) [reply internal/common/metrics/tags.go:88]: Sure! I would like to confirm the values before merging this (can add them in here if easiest).

[the remark this answers is not in the document]
Comment (cretz) [reply internal/common/metrics/tags.go:88]: So to confirm, it's `handler_error_NOT_IMPLEMENTED` not all consistent cased, correct? I don't mind, just checking.

[the remark this answers is not in the document]
Comment (bergundy) [reply internal/common/metrics/tags.go:88]: Yes, that's correct. Do you think it's worth doing a lower case conversion here?
I updated the docs PR to use upper case letters.

[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply internal/common/metrics/tags.go:88]: For gRPC errors the `status_code` tag is all caps

[the remark this answers is not in the document]
Comment (bergundy) [reply internal/common/metrics/tags.go:88]: Let's move on. I think we agree it's fine as long as it's consistent and documented.

**Pull request merged**